python-checks/src/test/resources/checks/confusingWalrus.py

```diff
@@ -0,0 +1,36 @@
+# using an assignment expression (:=) as an assignment statement (=) is more explicit
+(v := f(p))  # Noncompliant
+v0 = (v1 := f(p))  # Noncompliant
```

Comment:
what about the following?
```
x = 42 + (y:=2)
```

Reply:
After discussing it again with @nicolas-harraudeau-sonarsource, we will not raise issues in case of assignment expression within larger expression when in an assignment statement context.